packages/Webkul/Product/src/Type/AbstractType.php

```diff
@@ -775,9 +775,9 @@ public function prepareForCart($data)
                 'total_incl_tax'      => $convertedPrice * $data['quantity'],
                 'base_total'          => $price * $data['quantity'],
                 'base_total_incl_tax' => $price * $data['quantity'],
-                'weight'              => $this->product->weight ?? 0,
-                'total_weight'        => ($this->product->weight ?? 0) * $data['quantity'],
-                'base_total_weight'   => ($this->product->weight ?? 0) * $data['quantity'],
+                'weight'              => (int) ($this->product->weight ?? 0),
+                'total_weight'        => (int) ($this->product->weight ?? 0) * $data['quantity'],
+                'base_total_weight'   => (int) ($this->product->weight ?? 0) * $data['quantity'],
```

Comment:
All the three properties are decimal type, it should be cast as float.

Reply:
Okay, I will close this PR and create a new one based on branch 2.2. Thank you for your response!